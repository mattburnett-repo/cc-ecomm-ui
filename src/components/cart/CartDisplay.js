// TODO: cart should have its own state, to manage changes in line-item status like quanity / deletion

import NavBarDisplay from '../../components/nav/NavBarDisplay';

export default function CartDisplay ( props )  { // TODO: cartData should come from state / redux
    const { cartData } = props;
    const theCart = cartData[0].cart;

    return (
        <div>
            <NavBarDisplay calledFrom="cart" />
            <div role="presentation" aria-label="cart-header">
                cart_id: {theCart.cart_id} user_id: {theCart.user_id}
            </div>
            <div role="presentation" aria-label="cart-items">
                {theCart.cart_items.map((item, index) => (
                    <div role="presentation" aria-label="cart-item" key={index}>
                        <label htmlFor='product-id'>Product ID:</label>
                        <div id='product-id' aria-label="product-id">{item.product_id} </ div> 
                        <label htmlFor='product-name'>Product Name:</label>
                        <div id='product-name' aria-label="product-name">{item.product_name}</ div> 
                        <label htmlFor='product-price'>Product Price:</label>
                        <div id='product-price' aria-label="product-price">{item.product_price}</ div> 
                        <label htmlFor='product-quantity'>Product Quantity:</label>
                        <input id="product-quantity" role="textbox" aria-label='product-quantity' placeholder={item.product_quantity} />
                        <label htmlFor='product-total-price'>Product Total Price:</label>
                        <div id='product-total-price' aria-label="product-total-price">{item.product_total_price}</ div> 
                    </div>
                ))}     
            </div>
            <div role="presentation" aria-label="cart-total">
                <label htmlFor="cart-total-price">Cart Total:</label>
                <div id="cart-total-price" role="presentation" aria-label="cart-total-price">{theCart.cart_total_price.sum} </div>
            </div>
            <button aria-label="checkout-button">Checkout</button>
        </div>
    )    
}