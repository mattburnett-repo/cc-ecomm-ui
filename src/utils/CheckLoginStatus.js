
import { useSelector } from 'react-redux'
import { useHistory } from 'react-router-dom'

import { selectIsAuthenticated } from '../features/auth/authSlice'

export default function CheckLoginStatus () {
    // eslint-disable-next-line
    const isLoggedIn = useSelector(selectIsAuthenticated);

    const history = useHistory()

    if(!isLoggedIn) {
        history.push('/login')

        return false
    } else {
        return true;
    }
}        
