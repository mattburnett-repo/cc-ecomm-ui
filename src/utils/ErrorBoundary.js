import React from 'react'

export default class ErrorBoundary extends React.Component {
    constructor(props) {
      super(props);
      this.state = { hasError: false };
    }
  
    static getDerivedStateFromError(error) {
      // Update state to trigger fallback UI
      return { hasError: true };
    }
  
    componentDidCatch(error, errorInfo) {
        // Log error or send logging data to log management tool
        //   logErrorToMyService(error, errorInfo);
        console.log('ErrorBoundary ', errorInfo)
    }
  
    render() {
      if (this.state.hasError) {
        // Render fallback UI
        return <h1>There was an error. Sorry about that; we're working on it.</h1>;
      }
  
      return this.props.children; 
    }
  }

