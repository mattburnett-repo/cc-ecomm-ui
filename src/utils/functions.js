
// USAGE: formatAsCurrency.format(3.005); // "3.01"
const formatAsCurrency = new Intl.NumberFormat('en-US', {
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
});

const formatDate =   new Intl.DateTimeFormat("en-US", {
    year: "numeric",
    month: "long",
    day: "2-digit"
});

module.exports = {
    formatAsCurrency: formatAsCurrency,
    formatDate: formatDate
}