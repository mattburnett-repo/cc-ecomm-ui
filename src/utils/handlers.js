

// handler function declarations

// function handleAddToCart(e){ 
//     alert('handleAddToCart inside utils product_id: ' + e.target.id + ' quantity: ' + e.quantity)
//     // {() => dispatch(addItemToCurrentCart(id))}
//     // can't useDispatch here.
// } 

function handleCheckout(e){ 
    alert('handleCheckout inside utils ')
} 

function handleSearch(e){
    e.preventDefault()
    // let searchTerms = e.target.value
    alert('handleSearch for search term/s ' + e.target.searchTerms.value + '. \nneed to figure out how to get dispatch() to work from here')
}

function handleOrderClick(e) {
    e.preventDefault()

    let order_id = e.target.order_id.value
    alert('go to order detail component ' + order_id)
}

function handleHomeClick() {
    console.log('handleHomeClick')
}
function handleShowCartClick() {
    console.log('handleShowCartClick')
}
function handleCheckoutClick() {
    console.log('handleCheckoutClick')
}
function handleCancelClick() {
    console.log('handleCanceClick')
}


module.exports = {
    handleHomeClick: handleHomeClick,
    handleShowCartClick: handleShowCartClick,
    handleCheckoutClick: handleCheckoutClick,
    handleCancelClick: handleCancelClick,
    // handleAddToCart: handleAddToCart,
    handleCheckout: handleCheckout,
    handleSearch: handleSearch,
    handleOrderClick: handleOrderClick
} // end module.exports