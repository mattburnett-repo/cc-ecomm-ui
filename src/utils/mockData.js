// https://css-tricks.com/writing-tests-for-react-applications-using-jest-and-enzyme/

module.exports = {
    // someday I will figure out how to do this correctly...
    // getUserData: jest.fn(() => {
    //   return Promise.resolve({
    //   data: [
    //     {
    //       id: 1,
    //       username: 'asdf',
    //       email: 'asdf@asdf.com',
    //       password: 'asdf'
    //     }
    //   ]
    //   })
    // }),
    mockCartsData:[
        {
          "cart": {
              "cart_id": 1,
              "user_id": 1,
              "cart_items": [
                  {
                      "product_id": 1,
                      "product_name": "hedgehog",
                      "product_price": 1,
                      "product_quantity": 1,
                      "product_total_price": 1
                  },
                  {
                      "product_id": 2,
                      "product_name": "maine coon cat",
                      "product_price": 2,
                      "product_quantity": 2,
                      "product_total_price": 4
                  },
                  {
                      "product_id": 3,
                      "product_name": "bulldog",
                      "product_price": 3,
                      "product_quantity": 3,
                      "product_total_price": 9
                  },
                  {
                      "product_id": 4,
                      "product_name": "bunny",
                      "product_price": 4,
                      "product_quantity": 4,
                      "product_total_price": 16
                  }
              ],
              "cart_total_price": {
                  "sum": 30,
                  "cart_id": 1
              }
          }
      }
    ],
    mockCartData: {
      cart: {
        cart_id: 1,
        user_id: 1,
        cart_items: [
          {
            product_id: 1,
            product_name: 'hedgehog',
            product_price: 1,
            product_quantity: 1,
            product_total_price: 1
          },
          {
            product_id: 2,
            product_name: 'maine coon cat',
            product_price: 2,
            product_quantity: 2,
            product_total_price: 4
          },
          {
            product_id: 3,
            product_name: 'bulldog',
            product_price: 3,
            product_quantity: 3,
            product_total_price: 9
          },
          {
            product_id: 4,
            product_name: 'bunny',
            product_price: 4,
            product_quantity: 4,
            product_total_price: 16
          }
      ],
      cart_total_price: { sum: 30, cart_id: 1 }
    }},
    mockOrderData: [{
      "order": {
          "cart": {
              "cart_id": 1,
              "cart_items": [
                  {
                      "product_id": 1,
                      "product_name": "product_name_01",
                      "product_price": 1,
                      "product_quantity": 1,
                      "product_total_price": 1
                  },
                  {
                      "product_id": 2,
                      "product_name": "product_name_02",
                      "product_price": 2,
                      "product_quantity": 2,
                      "product_total_price": 4
                  },
                  {
                      "product_id": 3,
                      "product_name": "product_name_03",
                      "product_price": 3,
                      "product_quantity": 3,
                      "product_total_price": 9
                  },
                  {
                      "product_id": 4,
                      "product_name": "product_name_04",
                      "product_price": 4,
                      "product_quantity": 4,
                      "product_total_price": 16
                  }
              ]
          },
          "user_id": 1,
          "order_id": 1,
          "order_date": null,
          "total_price": {
              "sum": 30,
              "cart_id": 1
          }
        }
      }],
    mockOrderConfirmationData:
      {
        "order_id": 1,
        "total_amount": 1234,
        "order_date": '12-Dez-2021',
        "payment_type": 'Deutschmarks',
        "delivery_address": '123 Katzendamm, 90210 Berlin DE'
      },
    mockProductCategories: [
        // {
        //   "category_id": 0,
        //   "description": "Please select a product category"
        // },
        {
          "category_id": 4,
          "description": "Auto"
        },
        {
            "category_id": 3,
            "description": "Health"
        },
        {
            "category_id": 1,
            "description": "Pets"
        },
        {
            "category_id": 2,
            "description": "Tech"
        }
      ],
    mockProduct:[
        {
          "id": 2,
          "category_id": 2,
          "name": "product_name_02",
          "description": "product_desc_02",
          "image_url": "https://cdn.pixabay.com/photo/2015/03/27/13/16/maine-coon-694730_1280.jpg",
          "price": "2.00"
        },      
      ],
    mockProducts: [
        {
            "id": 1,
            "category_id": 1,
            "name": "product_name_01",
            "description": "product_desc_01",
            "image_url": "https://pixabay.com/images/id-71282/",
            "price": "1.00"
        },
        {
            "id": 2,
            "category_id": 2,
            "name": "product_name_02",
            "description": "product_desc_02",
            "image_url": "http://www.image.com/item/2",
            "price": "2.00"
        },
        {
            "id": 3,
            "category_id": 3,
            "name": "product_name_03",
            "description": "product_desc_03",
            "image_url": "http://www.image.com/item/3",
            "price": "3.00"
        },
        {
            "id": 4,
            "category_id": 4,
            "name": "product_name_04",
            "description": "product_desc_04",
            "image_url": "http://www.image.com/item/4",
            "price": "4.00"
        }
    ],
    mockUserData:[
      {
        "id": 4,
        "user_name": "asdf",
        "password": "$2b$10$OsTX1L59rYgsmAnsrJsXoO3RH3RSfLgEF53YBB3723B9eoELtUa.a",
        "email": "asdf@asdf.com",
        "google_id": null,
        "google_display_name": null,
        "google_first_name": null,
        "google_last_name": null,
        "google_image": null
      }
    ]
  } // end module.exports