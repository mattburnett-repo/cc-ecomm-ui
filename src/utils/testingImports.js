// export default function TestingImports() {
//     return (
//         import { render, screen, fireEvent } from '@testing-library/react';
//         import renderer from 'react-test-renderer';
//         import userEvent from '@testing-library/user-event';

//         import { Provider } from 'react-redux';
//         import store from '../../store'
//     )
// }