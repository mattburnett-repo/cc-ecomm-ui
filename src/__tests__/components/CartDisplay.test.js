// 20211217: Wireframe https://wireframe.cc/kXiUKu

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
// import userEvent from '@testing-library/user-event';

import CartDisplay from '../../components/cart/CartDisplay';

import { mockCartData } from '../../utils/mockData'

describe('<CartDisplay cartData={mockCartData} /> component tests', () => {
    beforeEach(() => render(<CartDisplay cartData={mockCartData} />));

    it('should render nav bar', () => {
      screen.getByRole('presentation', {name: /nav-bar/i});
    });
    it('should render cart items', () => {
        screen.getAllByRole('presentation', {name: /cart-items/i});
        screen.getAllByRole('textbox', { name: /product-quantity/i });
    });
    it('should render cart total price', () => {
        screen.getByRole('presentation', {name: /cart-total-price/i});
    });
    it('should render a checkout button', () => {
        screen.getByRole('button', {name: /checkout-button/i});
    });

    it('creates a snapshot', () => {
      const tree = renderer.create(<CartDisplay cartData={mockCartData} />).toJSON();
      expect(tree).toMatchSnapshot();  
  });
}); // end component
