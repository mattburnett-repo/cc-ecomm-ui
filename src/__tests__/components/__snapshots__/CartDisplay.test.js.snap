// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartDisplay cartData={mockCartData} /> component tests creates a snapshot 1`] = `
<div>
  <div
    aria-label="nav-bar"
    role="presentation"
  >
    <div
      aria-label="header-message"
      role="presentation"
    >
      cart
       
      header message
    </div>
    <button
      aria-label="home"
    >
      Home
    </button>
    <button
      aria-label="show-cart"
    >
      Show Cart
    </button>
    <button
      aria-label="checkout"
    >
      Checkout
    </button>
    <button
      aria-label="logout"
    >
      Log Out
    </button>
  </div>
  <div
    aria-label="cart-header"
    role="presentation"
  >
    cart_id: 
    1
     user_id: 
    1
  </div>
  <div
    aria-label="cart-items"
    role="presentation"
  >
    <div
      aria-label="cart-item"
      role="presentation"
    >
      <label
        htmlFor="product-id"
      >
        Product ID:
      </label>
      <div
        aria-label="product-id"
        id="product-id"
      >
        1
         
      </div>
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        id="product-name"
      >
        product_name_01
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        id="product-price"
      >
        1
      </div>
      <label
        htmlFor="product-quantity"
      >
        Product Quantity:
      </label>
      <input
        aria-label="product-quantity"
        id="product-quantity"
        placeholder={1}
        role="textbox"
      />
      <label
        htmlFor="product-total-price"
      >
        Product Total Price:
      </label>
      <div
        aria-label="product-total-price"
        id="product-total-price"
      >
        1
      </div>
    </div>
    <div
      aria-label="cart-item"
      role="presentation"
    >
      <label
        htmlFor="product-id"
      >
        Product ID:
      </label>
      <div
        aria-label="product-id"
        id="product-id"
      >
        2
         
      </div>
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        id="product-name"
      >
        product_name_02
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        id="product-price"
      >
        2
      </div>
      <label
        htmlFor="product-quantity"
      >
        Product Quantity:
      </label>
      <input
        aria-label="product-quantity"
        id="product-quantity"
        placeholder={2}
        role="textbox"
      />
      <label
        htmlFor="product-total-price"
      >
        Product Total Price:
      </label>
      <div
        aria-label="product-total-price"
        id="product-total-price"
      >
        4
      </div>
    </div>
    <div
      aria-label="cart-item"
      role="presentation"
    >
      <label
        htmlFor="product-id"
      >
        Product ID:
      </label>
      <div
        aria-label="product-id"
        id="product-id"
      >
        3
         
      </div>
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        id="product-name"
      >
        product_name_03
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        id="product-price"
      >
        3
      </div>
      <label
        htmlFor="product-quantity"
      >
        Product Quantity:
      </label>
      <input
        aria-label="product-quantity"
        id="product-quantity"
        placeholder={3}
        role="textbox"
      />
      <label
        htmlFor="product-total-price"
      >
        Product Total Price:
      </label>
      <div
        aria-label="product-total-price"
        id="product-total-price"
      >
        9
      </div>
    </div>
    <div
      aria-label="cart-item"
      role="presentation"
    >
      <label
        htmlFor="product-id"
      >
        Product ID:
      </label>
      <div
        aria-label="product-id"
        id="product-id"
      >
        4
         
      </div>
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        id="product-name"
      >
        product_name_04
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        id="product-price"
      >
        4
      </div>
      <label
        htmlFor="product-quantity"
      >
        Product Quantity:
      </label>
      <input
        aria-label="product-quantity"
        id="product-quantity"
        placeholder={4}
        role="textbox"
      />
      <label
        htmlFor="product-total-price"
      >
        Product Total Price:
      </label>
      <div
        aria-label="product-total-price"
        id="product-total-price"
      >
        16
      </div>
    </div>
  </div>
  <div
    aria-label="cart-total"
    role="presentation"
  >
    <label
      htmlFor="cart-total-price"
    >
      Cart Total:
    </label>
    <div
      aria-label="cart-total-price"
      id="cart-total-price"
      role="presentation"
    >
      30
       
    </div>
  </div>
  <button
    aria-label="checkout-button"
  >
    Checkout
  </button>
</div>
`;
